src/pudl_rmi/validate.py

```diff
@@ -0,0 +1,178 @@
+"""A module for data validation and QA/QC of the RMI outputs."""
+
+import logging
+from typing import List
+
+import numpy as np
+import pandas as pd
+import sqlalchemy as sa
+
+logger = logging.getLogger(__name__)
+
+
+RMI_TO_PUDL_COLS = {
+    "parent_name": "utility_name_parent",
+    "utility_name": "utility_name_ferc1",
+    "respondent_id": "utility_id_ferc1",
+    "year": "report_year",
+    "FERC_class": "ferc_acct_name",
+    "original_cost": "plant_balance_w_common",
+    "accum_depr": "book_reserve_w_common",
+    "net_plant_balance": "unaccrued_balance_w_common",
+}
+
+RMI_TO_PUDL_FERC_ACCT_NAME = {
+    "other_fossil": "Other",
+    "hydro": "Hydraulic",
+    "regional_transmission_and_market_operation": "Transmission",
+}
+
+
+def download_and_clean_net_plant_balance(pudl_engine: sa.engine.Engine) -> pd.DataFrame:
+    """
+    Download the Utility Transition Hub's cleaned FERC1 Net Plant Balance Table.
+
+    This function grabs the RMI Hub Team's cleaned version of the FERC1 Net
+    Plant Balance table. This function converts column names to PUDL column
+    names, standardizes FERC Account Names and merges in the ``utility_id_pudl``
+    through the ``utilities_ferc1`` PUDL table.
+
+    Note: In the fullness of time, this table should really come directly from
+    PUDL or the FERC1 database.
+
+    Args:
+        pudl_engine: A connection engine for the PUDL DB.
+    """
+    utils_f1 = pd.read_sql("utilities_ferc1", pudl_engine)
+    npb = pd.read_csv(
+        "https://utilitytransitionhub.rmi.org/static/data_download/net_plant_balance.csv"
+    )
+    npb = (
+        npb.convert_dtypes(convert_floating=False)
+        .rename(columns=RMI_TO_PUDL_COLS)
+        .replace({"ferc_acct_name": RMI_TO_PUDL_FERC_ACCT_NAME})
+        .assign(report_date=lambda x: pd.to_datetime(x.report_year, format="%Y"))
+        .merge(  # we need the pudl id!!
+            utils_f1[["utility_id_ferc1", "utility_id_pudl"]],
+            on=["utility_id_ferc1"],
+            how="left",
+            validate="m:1",
+        )
+    )
+    npb.loc[:, "ferc_acct_name"] = npb.ferc_acct_name.str.title()
+    return npb
+
+
+def add_data_source(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Add a data_source column with 'FERC' and 'PUC'.
+
+    The depreciation data includes a `data_source` column as a part of the
+    primary key of that data set. This is because there are sometimes we have
+    depreciation study from both `FERC` and `PUC` for the same utility and
```

Comment:
Copy edit.

Reply:
i tried... but now it is way long and idk if it is more legible.

Reply:
Sorry, I just meant this sentence is broken:
> This is because there are sometimes we have depreciation study from both...

I think it should be:
> This is because sometimes we have depreciation studies from both...